ibis/expr/datatypes.py

```diff
@@ -366,6 +372,42 @@ def can_implicit_cast(cls, other):
         return isinstance(other, (Floating, Decimal))
 
 
+_VALID_TIMEZONES = frozenset(pytz.all_timezones)
+
+
+@parametric
+class TimestampWithTimezone(DataType):
```

Comment:
Do you want this to be a different object versus the Arrow way (a single type with a timezone that can be None for tz-naive data)